agrirouter-api-java-api/src/main/java/com/dke/data/agrirouter/api/cancellation/DefaultCancellationToken.java

```diff
@@ -0,0 +1,34 @@
+package com.dke.data.agrirouter.api.cancellation;
+
+/** Default - time based - implementation of the cancellation token. */
+public class DefaultCancellationToken implements CancellationToken {
+
+  private int nrOfTries;
+  private final int maxTries;
+  private final long timeToWait;
+
+  public DefaultCancellationToken(int maxTries, long timeToWait) {
+    this.nrOfTries = 0;
+    this.maxTries = maxTries;
+    this.timeToWait = timeToWait;
+  }
+
+  @Override
+  public boolean isNotCancelled() {
+    return nrOfTries < maxTries;
+  }
+
+  @Override
+  public void step() {
+    nrOfTries++;
+  }
+
+  @Override
+  public void waitBeforeStartingTheNextStep() {
+    try {
+      Thread.sleep(timeToWait);
+    } catch (InterruptedException nop) {
+      // NOP
```

Comment:
An interrupted exception means, that the thread should be terminated as fast as possible and should continue its work. Therefore, in a normal implementation this would mean, that you wrap up the stuff that you are doing. In this case I would log the outcome and then terminate the token.

Reply:
Well, in that case, interrupting the Call-Loop is the goal of the exception, isn't it? The interuption exception is thrown when a parallel thread wants to interact with our thread. Terminating (=deleting?) the token would not be relevant as with the next call of the function it could be used again as the thread can work without interuption this time. Am I seeing something wrong here?